### ISSUE TITLE
[improve]: Login/Register user experience

### ISSUE THREAD

**Issue by kresimir-coko** [bytechefhq/bytechef#1695]
### Improvement target

We will use this issue to track the progress of the redesign and implementation of the user experience around logging in and registering a new account to Bytechef.

Link to [Figma](https://www.figma.com/design/rsUpnTsitNsSsaUH8zUk2c/bytechef-analiza?node-id=142-654&node-type=frame&t=S3SPBdoi3P5Bdih2-0)

## Goal

The new login/register ux should have the following:
- [x] Clear differentiation between login and register
- [x] Support for social media (Google & Github)
- [x] Forgotten password
- [x] Remember me checkbox
- [x] Add a feature flag to hide the Google and Github login options
- [ ] Add a feature flag to hide the Click to resend

### Improvement description

I think these are the components that currently implement this.

https://github.com/bytechefhq/bytechef/blob/master/client/src/pages/account/public/Login.tsx

https://github.com/bytechefhq/bytechef/blob/master/client/src/pages/account/public/Register.tsx

### Observed Version

master

**Comment by kresimir-coko** [comment]
@ivicac can you please expand upon the requirements for the login/register if I missed anything.

@akala444 I need you to post an update on the design after Ivica edits the ticket (or writes a new comment) so we can see if @katarina-cala can start working on it or not yet.

**Comment by ivicac** [comment]
@kresimir-coko requirements are ok; I left few comments in Figma

**Comment by aakarshitChoudhary** [comment]
@kresimir-coko , Hey is anyone working on this issue?, I like to work on it ?

**Comment by kresimir-coko** [comment]
Yes we are working on this atm @aakarshitChoudhary . We are in the middle of improving the UX, working with a designer so any most of tickets like these will be inhouse tickets.

I will try to open more tickets today so some new ones might fit your skillset. Thanks for reaching out! 
Keep a lookout for tickets with a `good first issue` label

**Comment by akala444** [comment]
I've updated the design based on the comments. @katarina-cala it's ready for development.
Link to [Figma](https://www.figma.com/design/rsUpnTsitNsSsaUH8zUk2c/bytechef-analiza?node-id=272-7007&t=fjp0DhVEI2ufX0uZ-1)

**Comment by kresimir-coko** [comment]
# Testing plan

This is not final - waiting on design to be finished.

## Login flow
- email + password + submit
- error testing
- clicking on google/github icon opens a corresponding window
- click on "forgot password"
- "forgot password" email submit
- "forgot password" click on "log in"
- show/hide password text

## Register flow
- email + password + submit
- clicking on google/github icon opens a corresponding window
- entering a valid email shows the password input
- show/hide password text
- password validation
- verify email opens when form submitted

**Comment by kresimir-coko** [comment]
@katarina-cala 
#Known bugs

- [ ] Email confirmation (security) text improvement
- [ ] New password - remove "duplicate password" text
- [ ] Create account error (security) text improvement

**Comment by kresimir-coko** [comment]
Can this ticket be closed? @katarina-cala @ivicac 
Do we have any more things to do for this to be considered done?

**Comment by akala444** [comment]
Hey, in my opinion there are a couple of things we should change:

- [ ] title - font-bold -> font-semibold
- [ ] background - surface-neutral-primary -> surface-neutral-secondary
- [ ] state (focused and hovered) for input field - color from black to brand
- [ ] password requirements (font size, icon size, gap between elements)

This the [link](https://app.zeplin.io/project/672c7c02f22eb42847135f12/screen/6735cca67427b3d67e12f82c) for Zeplin.

<img width="1007" alt="Screenshot 2025-01-11 at 14 49 50" src="https://github.com/user-attachments/assets/94f16e4a-dc74-4d83-ac53-bc6509a3e694" />

**Comment by kresimir-coko** [comment]
@ivicac please review @akala444's suggestions and give green light if they're good

**Comment by kresimir-coko** [comment]
Improvement to be done as part of https://github.com/bytechefhq/bytechef/issues/1922
